docs/v1/P4Runtime-Spec.mdk

```diff
@@ -1020,6 +1021,176 @@ table my_ipv4_lkup {
 }
 ~ End p4example
 
+### Structured Annotations
+
+P4 supports both unstructured and structured annotations [@P4Annotations].
+Unstructured annotations of the form `MyAnno1` or `MyAnno2(body-content)` can
+either be empty, or contain free-form content; anything between the pair of
+matched parentheses is legal. Conversely, structured annotations of the form
+`MyAnno3[]` or `MyAnno4[kvList|expressionList]` have a more prescribed syntax,
+which allows declaring key-value lists or expression lists. Both unstructured
+and structured annotations may be used simultaneously on a P4 element and
+`P4Info` supports this.
+
+The annotations described up to this point, &eg; `@brief()`, have all been
+unstructured annotations, or simply annotations. These are represented in
+`P4Info` as `repeated string annotations` fields in the various `message`s.
+Similarly, structured annotations are represented in `repeated
+StructuredAnnotation structured_annotations` fields which are siblings to the
+unstructured `annotations`. The `structured_annotations` contain parsed
+representations of the original annotation source. This parsing includes
+expression-evaluation, so the resulting `P4Info` may contain a simplified
+replica of the original structured annotations.
+
+The structured annotation messages are defined in p4types.proto.
+
+~ Begin Proto
+message KeyValuePair {
+  string key = 1;
+  Expression value = 2;
+}
+
+message KeyValuePairList {
+  repeated KeyValuePair kv_pairs = 1;
+}
+
+message Expression {
+  oneof value {
+    string string_value = 1;
+    int64 int64_value = 2;
+    bool bool_value = 3;
+  }
+}
+
+message ExpressionList {
+  repeated Expression expressions = 1;
+}
+
+message StructuredAnnotation {
+  string name = 1;
+  oneof body {
+    ExpressionList expression_list = 2;
+    KeyValuePairList kvpair_list = 3;
+  }
+}
+~ End Proto
+
+The `StructuredAnnotation` message can represent either a `KeyValuePairList`
+or an `ExpressionList`.
+
+The type of an expression is intentionally limited to one of the following
+base types: string literal, 64-bit signed integer, or boolean. The `p4c`
+compiler frontend which generates `P4Info` will evaluate all expressions and
+simplify them to one of the valid types. Any expressions which don't match one
+of the valid types will generate an error. For integers exceeding 64 bits,
+besides issuing an error, the compiler *could* print a suggestion to use a
```

Comment:
s/the compiler could/the compiler may

to follow RFC style

Reply:
Sure, I'll be happy to oblige both, thanks.